### PR TITLE
Rename tests to match naming convention and split up one test

### PR DESCRIPTION
**Pull request by RPangrle**

### CODE REVIEW

File: ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m [hunk 1]
```diff
@@ -54,35 +54,57 @@ - (void)tearDown
 }
 
 //Does not call the server, just passes invalid authority
-- (void)testValidateAuthorityError
+
+
+
+- (void)testValidateAuthority_whenSchemeIsHttp_shouldFailWithError
```
Comment (antrix1989) [inline ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m]: As far as I understand, the test name should be something like this testValidateAuthority_whenSchemeIsHttp_**shouldReturnValidatedFalseErrorNil**

According to our test doc:
> **Result** should describe the result of the method execution that test is checking with asserts

File: ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m [hunk 2]
```diff
@@ -54,35 +54,57 @@ - (void)tearDown
...
+         XCTAssertFalse(validated, @"\"%@\" should come back invalid.", authority);
+         XCTAssertNotNil(error);
+         
+         [expectation fulfill];
+     }];
+    
+    [self waitForExpectationsWithTimeout:1 handler:nil];
+}
+
+- (void)testValidateAuthority_whenAuthorityUrlIsInvalid_shouldFailWithError
```
Comment (antrix1989) [inline ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m]: Same here, testValidateAuthority_whenAuthorityUrlIsInvalid_shouldReturnValidatedFalseErrorNil

File: ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m [hunk 3]
```diff
@@ -54,35 +54,57 @@ - (void)tearDown
...
+         
+         [expectation fulfill];
+     }];
+    
+    [self waitForExpectationsWithTimeout:1 handler:nil];
 }
 
 // Tests a normal authority
-- (void)testAadNormalFlow
+- (void)testValidateAuthority_whenAuthorityValid_shouldPass
```
Comment (antrix1989) [inline ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m]: testValidateAuthority_whenAuthorityValid_shouldReturnValidatedTrueErrorNil

File: ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m [hunk 4]
```diff
@@ -109,7 +131,7 @@ - (void)testAadNormalFlow
 }
 
 //Ensures that an invalid authority is not approved
-- (void)testAadNonValidatedAuthority
+- (void)testValidateAuthority_whenAuthorityInvalid_shouldReturnError
```
Comment (antrix1989) [inline ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m]: testValidateAuthority_whenAuthorityInvalid_shouldReturnErrorValidatedFalse

File: ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m [hunk 5]
```diff
@@ -136,7 +158,7 @@ - (void)testAadNonValidatedAuthority
     XCTAssertFalse([authorityValidation isAuthorityValidated:[NSURL URLWithString:authority]]);
 }
 
-- (void)testBadAadAuthorityWithValidation
+- (void)testAcquireToken_whenAuthorityInvalid_shouldReturnError
```
Comment (antrix1989) [inline ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m]: testAcquireToken_whenAuthorityInvalid_shouldReturnErrorStatusFailed

File: ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m [hunk 6]
```diff
@@ -169,7 +191,7 @@ - (void)testBadAadAuthorityWithValidation
     [self waitForExpectationsWithTimeout:1 handler:nil];
 }
 
-- (void)testUnreachableAadServer
+- (void)testValidateAuthority_whenHostUnreachable_shouldFail
```
Comment (antrix1989) [inline ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m]: testValidateAuthority_whenHostUnreachable_shouldReturnErrorValidatedFalse

File: ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m (hunk 1 above)
Comment (oldalton) [inline ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m]: I personally think result can be described in different ways: e.g shouldFailWithError is a more generic result description and shouldReturnErrorValidatedFalse describes each of the results more precisely. Both are fine from my viewpoint.

**Comment by RPangrle** [comment]
Note that these are not unit tests, so the unit test naming convention does not hold. By design they test and validate multiple things.

File: ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m (hunk 1 above)
Comment (RPangrle) [inline ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m]: These aren't unit tests, by design they do multiple things and the unit testing conventions do not hold.

Comment (antrix1989) [inline ADAL/tests/integration/AADAuthorityValidationIntegrationTests.m]: Right, missed that it's integration test.

**Pull request merged**